probe-rs-target/Cargo.toml

```diff
@@ -11,14 +11,7 @@ categories = ["embedded", "hardware-support", "development-tools::debugging"]
 keywords = ["embedded"]
 license = "MIT OR Apache-2.0"
 
-[features]
-# serde(skip_serializing_if) breaks with `bincode`, because it doesn't encode metadata (field names).
-# Enable this if you want to serialize to bincode. Do not enable this if you want to serialize to other formats (YAML, JSON...)
-# https://github.com/serde-rs/serde/issues/1732
-# https://github.com/bincode-org/bincode/issues/326
-bincode = []
```

Comment:
why is this no longer required?

Reply:
It wasn't working anymore, and even in master it would generate null values.  Instead, in target-gen's main, I have added functionality to remove the Option::is-none values and extended this logic to include empty Vec as well as false bool values.  It makes for smaller yaml files and improves bincode generation times.

Reply:
I am a bit confused because bincode does not store metadata (field names). So how can this work with none values?

Reply:
Hmm odd. https://github.com/probe-rs/probe-rs/pull/1260 reports that this is an issue. Any idea what I am getting wrong here?

Reply:
@Yatekii I had similar options during development. I ended up doing the same as @mciantyre (also, serde-yaml was writing "null" values, rather than the "~" values I saw in older files) and turned of "Option::is_none" while I was working on other functionality, and then eventually settled on the option to strip the fields as target_gen writes the YAML files - I mostly went the home-made route because it gave me the added functionality of handling empty vecs and false bools also, while also reducing the number of feature flags in the repo.

Reply:
@burrbull , the `serde(default)` doesn't make a difference in this case, because the file is serialized to YAML with a value like `Field_X : null`. If it was omitted from the YAML like the `skip_serializing_if = "Option::is_none")` suggests, then the `default` would apply.

Reply:
the problem is not serialization from yaml but deserialization from bincode, which will only be apparent during runtime :)

Reply:
> Other places in the code this was used had it wrapped as part of a `#cfg_attr( not(feature=...` construct.

Thanks for the tip; I didn't notice that pattern. If we end up keeping the `bincode` feature, I'm happy to give it a try.